worlds/stardew_valley/stardew_rule/state.py

```diff
@@ -102,16 +105,19 @@ def value(self):
         return self.percent
 
     def __call__(self, state: CollectionState) -> bool:
-        stardew_world = state.multiworld.worlds[self.player]
+        stardew_world: "StardewValleyWorld" = state.multiworld.worlds[self.player]
         total_count = stardew_world.total_progression_items
         needed_count = (total_count * self.percent) // 100
         player_state = state.prog_items[self.player]
 
-        if needed_count <= len(player_state):
+        if needed_count <= len(player_state) - len(stardew_world.excluded_from_total_progression_items):
             return True
 
         total_count = 0
         for item, item_count in player_state.items():
+            if item in stardew_world.excluded_from_total_progression_items:
+                continue
+
             total_count += item_count
             if total_count >= needed_count:
                 return True
```

Comment:
I messages this in DMs, but any thoughts on this kind of thing:
```py
        player_state = {item: item_count for item, item_count in state.prog_items[self.player].items() if item not in stardew_world.excluded_from_total_progression_items}

        if needed_count <= len(player_state):
            return True

        if needed_count <= sum(player_state.values()):
            return True
```

Reply:
I'll benchmark it